tauri-app/src/routes/orders/add/+page.svelte

```diff
@@ -3,62 +3,98 @@
+  import { orderAdd, orderAddCalldata, orderAddComposeRainlang } from '$lib/services/order';
   import { ethersExecute } from '$lib/services/ethersTx';
   import { formatEthersTransactionError } from '$lib/utils/transaction';
+  import CodeMirrorRainlang from '$lib/components/CodeMirrorRainlang.svelte';
+  import { promiseTimeout } from '$lib/utils/time';
 
   let isSubmitting = false;
   let isCharting = false;
   let chartData: ChartData[];
   let dotrainFile = textFileStore('Rain', ['rain']);
   let deploymentRef: string | undefined = undefined;
+  let scenarioRef: string | undefined = undefined;
   let mergedConfigSource: ConfigSource | undefined = undefined;
   let mergedConfig: Config | undefined = undefined;
   let openAddOrderModal = false;
+  let rainlangText = "";
+  let resetRainlang = true;
 
   $: deployments = (mergedConfigSource !== undefined && mergedConfigSource?.deployments !== undefined && mergedConfigSource?.orders !== undefined) ?
     pickBy(mergedConfigSource.deployments, (d) => mergedConfig?.scenarios?.[d.scenario]?.deployer?.network?.name === $activeNetworkRef) : {};
   $: deployment = (deploymentRef !== undefined && mergedConfig !== undefined) ? mergedConfig.deployments[deploymentRef] : undefined;
   $: bindings = deployment ? deployment.scenario.bindings : {};
   $: $dotrainFile.text, updateMergedConfig();
 
-  $: rainlangExtension = new RawRainlangExtension({
-    hover: (text, position) => hoverCallback.apply(null, [text, position, bindings]),
-    diagnostics: async (text) => {
-      // get problems with merging settings config with frontmatter
-      const configProblems = await mergeDotrainConfigWithSettingsProblems(text.text);
-
-      // get problems with dotrain
-      const problems = await problemsCallback.apply(null, [text, bindings, deployment?.scenario.deployer.address]);
+  $: scenarios = (mergedConfigSource !== undefined && mergedConfigSource?.scenarios !== undefined) ?
+    pickBy(mergedConfigSource.scenarios, (d) => !d.deployer || mergedConfig?.deployers?.[d.deployer]?.network?.name === $activeNetworkRef) : {};
+  $: scenario = (scenarioRef !== undefined && mergedConfig !== undefined) ? mergedConfig.scenarios[scenarioRef] : undefined;
 
+  const rainlangExtension = new RawRainlangExtension({
+    diagnostics: async (text) => {
+      let configProblems = [];
+      let problems = []
+      try {
+        // get problems with merging settings config with frontmatter
+        configProblems = await mergeDotrainConfigWithSettingsProblems(text.text);
+      } catch(e) {
+        configProblems = [
+          {
+            msg: e as string,
+            position: [0, 0],
+            code: 9
+          }
+        ];
+      }
+      try {
+        // get problems with dotrain
+        problems = await promiseTimeout(
+          problemsCallback(text, bindings, deployment?.scenario.deployer.address),
+          5000,
+          "failed to parse on native parser"
+        );
+      } catch(e) {
```

Comment:
sentry

Reply:
this is not a sentry log, this is just a problem on lsp text area